src/api/include/migraphx/migraphx.hpp

```diff
@@ -1487,13 +1488,16 @@ quantize_int8(const program& prog, const target& ptarget, const quantize_int8_op
 
 struct experimental_custom_op_base
 {
-    virtual std::string name() const                                            = 0;
-    virtual argument compute(context ctx, shape output, arguments inputs) const = 0;
-    virtual shape compute_shape(shapes inputs) const                            = 0;
-    virtual std::vector<size_t> output_alias(shapes) const { return {}; }
+    experimental_custom_op_base()                                                             = default;
+    experimental_custom_op_base(experimental_custom_op_base const&)                           = default;
+    virtual ~experimental_custom_op_base()                                                    = default;
+
+    [[nodiscard]] virtual std::string name() const                                            = 0;
+    [[nodiscard]] virtual argument compute(context ctx, shape output, arguments inputs) const = 0;
+    [[nodiscard]] virtual shape compute_shape(shapes inputs) const                            = 0;
+    [[nodiscard]] virtual std::vector<size_t> output_alias(shapes) const { return {}; }
```

Comment:
`[[nodiscard]]` requires C++17. This header should work with at least C++11, so it should be removed.